sway/desktop/idle_inhibit_v1.c

```diff
@@ -0,0 +1,79 @@
+#include <stdlib.h>
+#include <wlr/types/wlr_idle.h>
+#include "log.h"
+#include "sway/desktop/idle_inhibit_v1.h"
+#include "sway/tree/view.h"
+#include "sway/server.h"
+
+
+static void handle_destroy(struct wl_listener *listener, void *data) {
+	struct sway_idle_inhibitor_v1 *inhibitor =
+		wl_container_of(listener, inhibitor, destroy);
+	wlr_log(L_DEBUG, "Sway idle inhibitor destroyed");
+	wl_list_remove(&inhibitor->link);
+	wl_list_remove(&inhibitor->destroy.link);
+	idle_inhibit_v1_check_active(inhibitor->manager);
+	free(inhibitor);
+}
+
+void handle_idle_inhibitor_v1(struct wl_listener *listener, void *data) {
+	struct wlr_idle_inhibitor_v1 *wlr_inhibitor = data;
+	struct sway_idle_inhibit_manager_v1 *manager =
+		wl_container_of(listener, manager, new_idle_inhibitor_v1);
+	wlr_log(L_DEBUG, "New sway idle inhibitor");
+
+	struct sway_idle_inhibitor_v1 *inhibitor =
+		calloc(1, sizeof(struct sway_idle_inhibitor_v1));
+	if (!inhibitor) {
+		return;
+	}
+
+	inhibitor->manager = manager;
+	inhibitor->view = view_from_wlr_surface(wlr_inhibitor->surface);
+	wl_list_insert(&manager->inhibitors, &inhibitor->link);
+
+
+	inhibitor->destroy.notify = handle_destroy;
+	wl_signal_add(&wlr_inhibitor->events.destroy, &inhibitor->destroy);
+
+	idle_inhibit_v1_check_active(manager);
+}
+
+void idle_inhibit_v1_check_active(
+		struct sway_idle_inhibit_manager_v1 *manager) {
+	struct sway_idle_inhibitor_v1 *inhibitor;
+	bool inhibited = false;
+	wl_list_for_each(inhibitor, &manager->inhibitors, link) {
+		if (!inhibitor->view) {
+			/* Cannot guess if view is visible so assume it is */
+			inhibited = true;
+			break;
+		}
+		if (view_is_visible(inhibitor->view)) {
+			inhibited = true;
+			break;
+		}
+	}
+	wlr_idle_set_enabled(manager->idle, NULL, !inhibited);
```

Comment:
This seems like an odd way to go about it. What if there are multiple inhibitors?

Reply:
That's precisely what this code handles: if any inhibitor has a visible view, the wlr_idle will be inhibited (the argument for `wlr_idle_set_enabled` is `enabled` so `!inhibited`)
If no inhibitors are on visible views then `inhibited` will have its default false value and idle timers will be reactivated.

Reply:
Wow I'm dumb